src/core/internal/hash.d

```diff
@@ -725,6 +728,84 @@ private size_t bytesHash(bool dataKnownToBeAligned)(scope const(ubyte)[] bytes,
     return h1;
 }
 
+static if (size_t.sizeof == 8)
+@nogc nothrow pure @trusted
+private ulong bytesHash64()(scope const ubyte[] bytes, ulong seed)
+{
+    alias h1 = seed;
+
+    enum ulong c1 = 0x87c37b91114253d5;
+    enum ulong c2 = 0x4cf5ad432745937f;
+    enum uint  c3 = 0x52dce729;
+
+    const(ubyte)* data = bytes.ptr;
+    //----------
+    // body
+    for (const end_data = bytes.ptr + (bytes.length & ~7);
+         data !is end_data;
+         data += 8)
+    {
+        version (BigEndian)
+        {
+            auto k1 =
+                   ((cast(ulong) data[0]) << 56) |
+                   ((cast(ulong) data[1]) << 48) |
+                   ((cast(ulong) data[2]) << 40) |
+                   ((cast(ulong) data[3]) << 32) |
+                   ((cast(ulong) data[4]) << 24) |
+                   ((cast(ulong) data[5]) << 16) |
+                   ((cast(ulong) data[6]) << 8)  |
+                    (cast(ulong) data[7]);
+        }
```

Comment:
Couldn't this use the pointer read optimization as done for the 32 bit hash above?

Reply:
Now it does.